src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -0,0 +1,353 @@
+  FoxFarmingEarnOpportunityType,
+  lpOpportunity,
+} from './foxEthCommon'
+import { fetchPairData } from './utils'
+
+type FoxEthState = {
+  farmOpportunities: FoxFarmingEarnOpportunityType[]
+  lpOpportunity: FoxEthLpEarnOpportunityType
+}
+
+const initialState: FoxEthState = {
+  farmOpportunities,
+  lpOpportunity,
+}
+
+const moduleLogger = logger.child({ namespace: ['foxEthSlice'] })
+
+export const foxEth = createSlice({
+  name: 'foxEth',
+  initialState,
+  reducers: {
+    clear: () => initialState,
+    upsertLpOpportunity: (state, action: PayloadAction<Partial<FoxEthLpEarnOpportunityType>>) => {
+      state.lpOpportunity = {
+        ...state.lpOpportunity,
+        ...action.payload,
+      }
+    },
+    upsertFarmingOpportunity: (
+      state,
+      action: PayloadAction<Partial<FoxFarmingEarnOpportunityType>>,
+    ) => {
+      const stateOpportunityIndex = state.farmOpportunities.findIndex(
+        opportunity => opportunity.contractAddress === action.payload.contractAddress,
+      )
+      state.farmOpportunities[stateOpportunityIndex] = {
+        ...state.farmOpportunities[stateOpportunityIndex],
+        ...action.payload,
+      }
+    },
+  },
+})
+
+type getFoxEthLpGeneralDataReturn = {
+  tvl: string
+  apy: string
+  isLoaded: boolean
+}
+type getFoxEthLpWalletDataReturn = {
+  underlyingFoxAmount: string
+  underlyingEthAmount: string
+  cryptoAmount: string
+  fiatAmount: string
+}
+
+type GetFoxFarmingContractGeneralDataReturn = {
+  expired: boolean
+} & getFoxEthLpGeneralDataReturn
+
+type GetFoxFarmingContractGeneralDataArgs = {
+  contractAddress: string
+}
+
+type getFoxFarmingContractWalletDataReturn = {
+  cryptoAmount: string
+  fiatAmount: string
+  unclaimedRewards: string
+}
+
+type getFoxFarmingContractWalletDataArgs = {
+  contractAddress: string
+  ethWalletAddress: string
+}
+
+type getFoxEthLpWalletDataArgs = {
+  ethWalletAddress: string
+}
+
+export const foxEthApi = createApi({
+  reducerPath: 'foxEthApi',
+  // not actually used, only used to satisfy createApi, we use a custom queryFn
+  baseQuery: fetchBaseQuery({ baseUrl: '/' }),
+  // refetch if network connection is dropped, useful for mobile
+  refetchOnReconnect: true,
+  endpoints: build => ({
+    getFoxEthLpGeneralData: build.query<getFoxEthLpGeneralDataReturn, void>({
+      queryFn: async (_args, injectedStore) => {
+        try {
+          const { getState, dispatch } = injectedStore
+          const state: any = getState() // ReduxState causes circular dependency\
+          const ethPrecision = state.assets.byId[ethAssetId].precision
+          const lpAssetPrecision = state.assets.byId[foxEthLpAssetId].precision
+          const ethPrice = state.marketData.crypto.byId[ethAssetId].price
+          const ethersProvider = getEthersProvider()
+          const uniV2LPContract = new Contract(
+            UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+            IUniswapV2Pair.abi,
+            ethersProvider,
+          )
```

Comment:
Does `new Contract()` make an infura request? If so we might want to singleton it